code/game/objects/structures/cannons/mounted_guns/mounted_gun.dm

```diff
@@ -0,0 +1,187 @@
+	icon_state = "pipeorgangun"
+	icon_state_base = "pipeorgangun"
+	icon_state_fire = "pipeorgangun_fire"
+	anchored = FALSE
+	anchorable_gun = TRUE
+	max_shots_per_fire = 8
+	shots_in_gun = 8
+	shots_per_load = 2
+	ammo_type = /obj/item/ammo_casing/junk
+	projectile_type = /obj/projectile/bullet/junk
+	loaded_gun = TRUE
+	fully_loaded_gun = TRUE
+	fire_delay = 3
+	shot_delay = 2
+	firing_shakes_camera = FALSE
+
+/obj/structure/mounted_gun/pipe/examine_more(mob/user)
+	. = ..()
+	. += span_notice("<b><i>Looking down at the [name], you recall a tale told to you in some distant memory...</i></b>")
+
+	. += span_info("To commit an act of vengeance is not unlike to enter a blood pact with a devil, ending the life of another, at the cost of your own.")
+	. += span_info("When humanity first spilled the blood of its own kind, with likely nothing more than a rock, the seal was broken. Vengeance was borne unto the world.")
+	. += span_info("However, vengeance alone is not enough to carry through the grim deed of murder. One must an gain advantage over their adversary.")
+	. += span_info("As such, the man who ended another's life with a stone, was in turn smote himself by another wielding a spear. After spears, bows. Swords. Guns. Tanks. Missiles. And on and on Vengeance fed. Growing stronger. Growing Worse.")
+	. += span_info("Vengeance persists to this day. It sometimes may slumber, seemingly content with having gorged itself, but in the end, its ceaseless hunger can be neither numbed nor sated.")
+
+/obj/structure/mounted_gun/pipe/fire()
+	if (!loaded_gun)
+		balloon_alert_to_viewers("Gun is not loaded!","",2)
+		return
+	for(var/times_fired = 1, times_fired <= shots_in_gun, times_fired++) //The normal DM for loop structure since the times it has fired is changing in the loop itself.
+		for(var/mob/shaken_mob in urange(10, src))
+			if((shaken_mob.stat == CONSCIOUS)&&(firing_shakes_camera == TRUE))
+				shake_camera(shaken_mob, 3, 1)
+			icon_state = icon_state_fire
+		if(loaded_gun)
+			playsound(src, fire_sound, 50, TRUE)
+
+			var/list_of_projectiles = list(
+			/obj/projectile/bullet/junk = 40,
+			/obj/projectile/bullet/incendiary/fire/junk = 25,
+			/obj/projectile/bullet/junk/shock = 25,
+			/obj/projectile/bullet/junk/hunter = 20,
+			/obj/projectile/bullet/junk/phasic = 8,
+			/obj/projectile/bullet/junk/ripper = 8,
+			/obj/projectile/bullet/junk/reaper = 3,
+			)
+			projectile_type = pick_weight(list_of_projectiles)
+
+			var/obj/projectile/fired_projectile = new projectile_type(get_turf(src))
+			fired_projectile.firer = src
+			fired_projectile.fired_from = src
+			fired_projectile.fire(dir2angle(dir))
+		sleep(shot_delay)
+	loaded_gun = FALSE
+	shots_in_gun = 0
+	fully_loaded_gun = FALSE
+	icon_state = icon_state_base
+
+/obj/structure/mounted_gun/canister_gatling //for the funny skeleton pirates! (WIP)
```

Comment:
still WIP?

Reply:
Originally, this was a test item to mess around with the mounted gun class, that I liked too much to cut.